src/bio/bio_config.c

```diff
@@ -334,7 +335,12 @@ load_bdev_subsystem_config(struct json_config_ctx *ctx, bool vmd_enabled,
 	while (key != NULL) {
 		if (spdk_json_strequal(key, "traddr")) {
 			traddr = spdk_json_strdup(json_value(key));
-
+			if (!traddr) {
+				D_ERROR("Failed to allocate memory");
+				rc = -DER_NOMEM;
+				goto out;
+			}
+			snprintf(traddr_new, SPDK_NVMF_TRADDR_MAX_LEN, "%s", traddr);
```

Comment:
could just snprintf() to taddr_new from json_value(key) directly and completely remove use of the traddr pointer (and strdup